bus-mapping/src/circuit_input_builder.rs

```diff
@@ -343,9 +376,54 @@ impl<'a, C: CircuitsParams> CircuitInputBuilder<C> {
     }
 
     /// set chunk context
-    pub fn set_chunkctx(&mut self, chunk_ctx: ChunkContext) {
+    pub fn set_chunk_ctx(&mut self, chunk_ctx: ChunkContext) {
+        self.chunks = (0..chunk_ctx.total_chunks)
```

Comment:
it's not very efficient that
1. self.chunks was setup in `new`
2. then the original setup is discard and re-instantiate in set_chunk_ctx
3. there are quite of code duplication